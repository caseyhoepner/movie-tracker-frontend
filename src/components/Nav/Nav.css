.nav-container {
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  width: 200px;
  font-size: 1.25rem;
}

.fav-btn-group {
  /*text should be display: none until :hover, then CSS transition ease in above the respective icon;*/
}

.fav-btn-nav {
  height: 50px;
  width: 50px;
  background-image: url('../../assets/images/hatFull-red.svg');
  background-size: cover;
}

.num-favs {
  position: relative;
  top: 5px;
  left: 25px;
}

.favorites-text,
.login-text {
  color: #D4A931;
  visibility: hidden;
}

.active p {
  position: relative;
  bottom: 45px;
  right: 50px;
  visibility: visible;
  transition: visibility 2s ease-in-out;
}

.active.login-text {
  bottom: 45px;
  right: 50px;
}

/*.active.favorites-text {
  bottom: 60px;
  right: 40px;
}*/

.login-btn {
  width: 100px;
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.login-icon {
  height: 50px;
  background-image: url('../../assets/images/margot.png');
  background-size: cover;
}

a {
  color: #D4A931;
  text-decoration: none;
}