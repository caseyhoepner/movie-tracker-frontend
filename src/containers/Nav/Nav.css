.nav-container {
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  width: 200px;
  font-size: 1.25rem;
}

.fav-btn-group {
  /*text should be display: none until :hover, then CSS transition ease in above the respective icon;*/
}

.fav-btn-nav {
  height: 50px;
  width: 50px;
  background-image: url('../../assets/images/hatFull-red.svg');
  background-size: cover;
}

.num-favs {
  position: relative;
  top: 5px;
  left: 25px;
}

.favorites-text,
.login-text {
  position: relative;
  color: #D4A931;
  opacity: 0;
}

a:hover p {
  opacity: 1;
}

a:hover .login-text {
  bottom: 50px;
  right: 50px;
}

a:hover .favorites-text {
  bottom: 50px;
  right: 25px;
}

.login-text {
  bottom: 30px;
  right: 50px;
  transition: 1s;
}

.favorites-text {
  bottom: 30px;
  right: 25px;
  transition: 1s;
}

.login-btn {
  width: 100px;
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.login-icon {
  height: 50px;
  background-image: url('../../assets/images/margot.png');
  background-size: cover;
}

a {
  color: #D4A931;
  text-decoration: none;
}