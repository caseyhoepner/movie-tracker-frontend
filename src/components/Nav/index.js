import React from 'react';
import PropTypes from 'prop-types';
import './Nav.css';
import { NavLink } from 'react-router-dom';
import userIcon from '../../assets/images/margot.png'

const Nav = () => {
  return (
    <nav className="nav-container">
      <NavLink
        className="fav-btn-group"
        to="/favorites">
        <div className="fav-btn-nav">
          <span className="num-favs">0</span>
        <p className="favorites-text">FAVORITES</p>
        </div>
      </NavLink>
      <NavLink to="/login" className="login-btn">
        <img
          className="login-icon"
          src={ userIcon } />
        <p className="login-text">LOGIN</p>
      </NavLink>
    </nav>
  )
}

export default Nav;