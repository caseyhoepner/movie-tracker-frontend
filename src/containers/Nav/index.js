import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { removeUser, filterMovies } from '../../actions';
import userIcon from '../../assets/images/margot.png';
import './Nav.css';

export const Nav = (props) => {
  const { userLoggedIn, removeUser, movies, filterMovies } = props

  const handleHover = (event) => {
    if(!event.target.classList.contains('active')) {
      event.target.classList.add('active')
    } else {
      event.target.classList.remove('active')
    }
  }

  return (
    <nav className="nav-container">
      <NavLink
        onClick={() => filterMovies(movies)}
        to="/favorites"
        className="fav-btn-group"
      >
        <div
          className="fav-btn-nav"
          onMouseOver={handleHover}
          onMouseOut={handleHover}
        >
          <span className="num-favs">0</span>
          <p
            className={`favorites-text`}
          >
            FAVORITES
          </p>
        </div>
      </NavLink>
      <NavLink
        to="/login"
        className={userLoggedIn ? "hidden" : "show"}
      >
        <div
          className="login-btn"
          onMouseOver={handleHover}
          onMouseOut={handleHover}
        >
          <img
            className="login-icon"
            src={ userIcon }
            alt='Click here to login'/>
          <p className="login-text">
            LOGIN
          </p>
        </div>
      </NavLink>
      <NavLink
        className={userLoggedIn ? "logout-btn" : "hidden"}
        onClick={removeUser}
        to='/'
      >
        <img
          className="login-icon"
          src={ userIcon }
          alt='Click here to login'/>
        <p className="login-text">
          LOGOUT
        </p>
      </NavLink>
    </nav>
  )
}

export const mapStateToProps = (state) => ({
  userLoggedIn: state.user.id,
  movies: state.movies
})

export const mapDispatchToProps = (dispatch) => ({
  removeUser: () => dispatch(removeUser()),
  filterMovies: (movies) => dispatch(filterMovies(movies))
})

Nav.propTypes = {
  removeUser: PropTypes.func.isRequired,
  userLoggedIn: PropTypes.number
}

export default connect(mapStateToProps, mapDispatchToProps)(Nav);