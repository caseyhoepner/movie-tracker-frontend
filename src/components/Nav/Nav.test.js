import React from 'react';
import { shallow } from 'enzyme';
import Nav from './index.js';

describe('Nav', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Nav />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})

