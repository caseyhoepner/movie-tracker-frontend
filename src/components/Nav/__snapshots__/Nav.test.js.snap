// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
<nav>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/favorites"
  >
    <div
      className="fav-btn"
    >
      <span
        className="num-favs"
      >
        0
      </span>
    </div>
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="login-btn"
    to="/login"
  >
    <img
      className="login-icon"
      src=""
    />
    <p
      className="login-text"
    />
  </NavLink>
</nav>
`;
